lib/offline/storage.js

```diff
@@ -770,6 +788,61 @@ shaka.offline.Storage.prototype.createOfflineManifest_ = function(
   };
 };
 
+/**
+ * Extract DRM initData from first segment
+ *
+ * @param {shaka.extern.Manifest} manifest
+ * @param {shaka.net.NetworkingEngine} networkingEngine
+ * @return {Promise<shaka.extern.Manifest>}
+ * @private
+ */
+shaka.offline.Storage.prototype.extractInitDataFromFirstSegment_ = async function(
+  manifest, networkingEngine) {
+  // get init segments uri of all variants
+  /** @type {Array<Array<string>>} */
+  let initSegmentReferences = [].concat.apply([], manifest.periods.map((period) => {
+    return period.variants.map((variant) => {
+      return variant.video.initSegmentReference;
+    });
+  }));
+
+  // download init segments
+  let segmentResult = await Promise.all(
+    initSegmentReferences.map((initSegmentReference) => {
+      this.checkDestroyed_();
+      let retryParams = this.player_.getConfiguration()
+          .streaming.retryParameters;
+      let request = shaka.net.NetworkingEngine.makeRequest(
+        initSegmentReference.getUris(), retryParams);
+      if (initSegmentReference.startByte != null) {
+        let end = (initSegmentReference.endByte != null ?
+          initSegmentReference.endByte : '');
+        request.headers['Range'] = 'bytes=' + initSegmentReference.startByte
+          + '-' + end;
+      }
+      let type = shaka.net.NetworkingEngine.RequestType.SEGMENT;
+      return networkingEngine.request(type, request).promise;
+    })
+  );
+
+  // search for pssh atom in segment and insert init data into manifest
+  let counter = 0;
+  manifest.periods.forEach((period) => {
```

Comment:
Our standards have been updated to avoid `forEach` unless necessary. Please update your code to use the `for (const x of y) { ... }` syntax.

Reply:
we can update to the latest shaka standards